packages/tools/src/utilities/livewire/LivewireScissors.ts

```diff
@@ -0,0 +1,589 @@
+import { Types } from '@cornerstonejs/core';
+import { BucketQueue } from '../BucketQueue';
+
+const MAX_UINT32 = 4294967295;
+const TWO_THIRD_PI = 2 / (3 * Math.PI);
+
+/**
+ * Scissors
+ *
+ * Ref: Eric N. Mortensen, William A. Barrett, Interactive Segmentation with
+ *   Intelligent Scissors, Graphical Models and Image Processing, Volume 60,
+ *   Issue 5, September 1998, Pages 349-384, ISSN 1077-3169,
+ *   DOI: 10.1006/gmip.1998.0480.
+ *
+ * {@link http://www.sciencedirect.com/science/article/B6WG4-45JB8WN-9/2/6fe59d8089fd1892c2bfb82283065579}
+ *
+ * Implementation based on
+ * {@link http://code.google.com/p/livewire-javascript/}
+ */
+export class LivewireScissors {
+  private searchGranularityBits: number;
+  private searchGranularity: number;
+
+  /** Width of the image */
+  private width: number;
+
+  /** Height of the image */
+  private height: number;
+
+  /** Grayscale image */
+  private grayscalePixelData: Float32Array;
+
+  // Laplace zero-crossings (either 0 or 1).
+  private laplace: Float32Array;
+
+  /** Gradient vector magnitude for each pixel */
+  private gradMagnitude: Float32Array;
+
+  /** Gradient of each pixel in the x-direction */
+  private gradXNew: Float32Array;
+
+  /** Gradient of each pixel in the y-direction */
+  private gradYNew: Float32Array;
+
+  /** Dijkstra - start point */
+  private startPoint: Types.Point2;
+
+  /** Dijkstra - store the state of a pixel (visited/unvisited) */
+  private visited: boolean[];
+
+  /** Dijkstra - map a point to its parent along the shortest path to root (start point) */
+  private parents: Uint32Array;
+
+  /** Dijkstra - store the cost to go from the start point to each node */
+  private costs: Float32Array;
+
+  /** Dijkstra - BucketQueue to sort items by priority */
+  private priorityQueueNew: BucketQueue<number>;
+
+  constructor(grayscalePixelData: Float32Array, width: number, height: number) {
+    const numPixels = grayscalePixelData.length;
+
+    this.searchGranularityBits = 8; // Bits of resolution for BucketQueue.
+    this.searchGranularity = 1 << this.searchGranularityBits; //bits.
+
+    this.width = width;
+    this.height = height;
+
+    this.grayscalePixelData = grayscalePixelData;
+    this.laplace = null;
+    this.gradXNew = null;
+    this.gradYNew = null;
+
+    this.laplace = this._computeLaplace();
+    this.gradMagnitude = this._computeGradient();
+    this.gradXNew = this._computeGradientX();
+    this.gradYNew = this._computeGradientY();
+
+    this.visited = new Array(numPixels);
+    this.parents = new Uint32Array(numPixels);
+    this.costs = new Float32Array(numPixels);
+  }
+
+  public startSearch(startPoint: Types.Point2): void {
+    const startPointIndex = this._getPointIndex(startPoint[1], startPoint[0]);
+
+    this._reset();
```

Comment:
Now that `reset()` is called only by this function it makes sense to add that code here and delete `reset()`